scripts/shared.sh

```diff
@@ -4,11 +4,94 @@
 set -a
 set -eu -o pipefail
 
+function _extract_from_file() {
+    local key="$1"
+    local config_file="$2"
+    local value
+    if [[ -f $config_file ]]; then
+        value="$(eval "jq -r .${key} $config_file")"
```

Comment:
why is this eval necessary?

Reply:
hmm not sure! this is how I got it to work hehe, my bash-fu is admittedly weak, but `jq` was complaining about ${key}

Reply:
I'm also intrigued by the apparent necessity for the `eval`

This looks like you might be trying to support deep key lookup. I think this could all be one-lined to

```
jq -er ".$key // empty" "$config_file"
```
You won't need to check the return `$value` anymore. Furthermore, callers of this function can use the exit status to determine success/failure, so in theory you could rewrite `_extract_cluster_config_value()` to invoke those three queries in short-circuiting fashion (`a || b || c`)

Reply:
so using `jq -er ".$key // empty" "$config_file"` did not work because `false` values would end up as `empty`, I did however clean this up a bunch and removed the use of `eval`